README.md

```diff
@@ -54,6 +54,10 @@ You need the following tools to build the library with the provided makefiles:
 * Microsoft Visual Studio 2013 or later (if using Visual Studio).
 * Doxygen 1.8.11 or later (if building the documentation; slightly older versions should work).
 
+### Git usage
+
+The `development` branch and the `mbedtls-3.6` long-term support of Mbed TLS use a [Git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules#_cloning_submodules) ([framework](https://github.com/Mbed-TLS/mbedtls-framework)). This is not needed to merely compile the library at a release tag. This is not needed to consume a release archive (zip or tar).
```

Comment:
Do we want to keep committing the generated files and now the whole submodule at each release tag?
Since we are giving up on the autogenerated tarballs anyways, we should probably update the release process instead of removing and re-adding the submodule for each release.

Reply:
Or did you mean that we will keep committing the generated files for now, and that makes the files in the submodule unnecessary?

Reply:
For 3.6.x the current plan is to keep committing the generated files, yes. So `git clone mbedtls mbedtls-3.6.0` (without `--recurse-submodules`) would give you enough to do `make lib` but not enough to do `make test`.

Our tarballs will include the submodule content.